src/form/en_ivr.jl

```diff
@@ -0,0 +1,1763 @@
+    vi_to = var(pm, nw, :vi, t_bus)
+
+    crt_fr = var(pm, nw, :crt, f_idx)
+    cit_fr = var(pm, nw, :cit, f_idx)
+    crt_to = var(pm, nw, :crt, t_idx)
+    cit_to = var(pm, nw, :cit, t_idx)
+
+    if config==WYE || length(crt_fr)==1
+        P_fr = f_connections[1:end-1]
+        n_fr = f_connections[end]
+        vrt_fr = [vr_fr[p]-vr_fr[n_fr] for p in P_fr]
+        vit_fr = [vi_fr[p]-vi_fr[n_fr] for p in P_fr]
+    elseif config==DELTA && length(crt_fr)==3
+        M = _get_delta_transformation_matrix(3)
+        vrt_fr = M*[vr_to[p] for p in f_connections]
+        vit_fr = M*[vi_to[p] for p in f_connections]
+    else
+        error("The configuration $config of dimension $(length(crt)) is not supported.")
+    end
+
+    P_to = t_connections[1:end-1]
+    n_to = t_connections[end]
+    vrt_to = [vr_to[p]-vr_to[n_to] for p in P_to]
+    vit_to = [vi_to[p]-vi_to[n_to] for p in P_to]
+
+    idxs = [1:length(vrt_fr)...]
+    pt_fr = JuMP.@NLexpression(pm.model, [i in idxs],  vrt_fr[i]*crt_fr[i] + vit_fr[i]*cit_fr[i])
+    qt_fr = JuMP.@NLexpression(pm.model, [i in idxs], -vrt_fr[i]*cit_fr[i] + vit_fr[i]*crt_fr[i])
+    pt_to = JuMP.@NLexpression(pm.model, [i in idxs],  vrt_to[i]*crt_to[i] + vit_to[i]*cit_to[i])
+    qt_to = JuMP.@NLexpression(pm.model, [i in idxs], -vrt_to[i]*cit_to[i] + vit_to[i]*crt_to[i])
+
+    if sm_ub<Inf
+        JuMP.@NLconstraint(pm.model, sum(pt_fr[i] for i in idxs)^2 + sum(qt_fr[i] for i in idxs)^2 <= sm_ub^2)
+        JuMP.@NLconstraint(pm.model, sum(pt_to[i] for i in idxs)^2 + sum(qt_to[i] for i in idxs)^2 <= sm_ub^2)
+    end
+
+    if report
+        sol(pm, nw, :transformer, id)[:pf] = pt_fr
+        sol(pm, nw, :transformer, id)[:qf] = qt_fr
+        sol(pm, nw, :transformer, id)[:pt] = pt_to
+        sol(pm, nw, :transformer, id)[:qt] = qt_to
+        sol(pm, nw, :transformer, id)[:smtot_fr] = JuMP.@NLexpression(pm.model, sqrt(sum(pt_fr[i] for i in idxs)^2 + sum(qt_fr[i] for i in idxs)^2))
+        sol(pm, nw, :transformer, id)[:smtot_to] = JuMP.@NLexpression(pm.model, sqrt(sum(pt_to[i] for i in idxs)^2 + sum(qt_to[i] for i in idxs)^2))
+    end
+end
+
+
+# TRANSFORMER - Constraint - Quadratic
+
+"""
+	function constraint_mc_transformer_power_rating(
+		pm::AbstractQuadraticExplicitNeutralIVRModel,
+		nw::Int,
+		id::Int,
+		f_idx::Tuple,
+		t_idx::Tuple,
+		f_bus::Int,
+		t_bus::Int,
+		f_connections::Vector,
+		t_connections::Vector,
+		config::ConnConfig,
+		sm_ub::Real;
+		report::Bool=true
+	)
+
+For quadratic IVR models with explicit neutrals,
+imposes a bound on the magnitude of the total apparent power at both windings.
+
+```
+sum(pt_fr)^2 + sum(qt_fr)^2 <= sm_ub^2
+sum(pt_to)^2 + sum(qt_to)^2 <= sm_ub^2
+```
+"""
+function constraint_mc_transformer_power_rating(pm::AbstractQuadraticExplicitNeutralIVRModel, nw::Int, id::Int, f_idx::Tuple, t_idx::Tuple, f_bus::Int, t_bus::Int, f_connections::Vector, t_connections::Vector, config::ConnConfig, sm_ub::Real; report::Bool=true)
```

Comment:
Should be `thermal_rating` to be consistent with other naming conventions

Reply:
You mean `thermal_limit` right?